### PR TITLE
Added credentials to the author byline in article's author fields

### LINKED ISSUE #5171: Person credentials are not displayed on teasers or bylines
# Story

As a content editor, I can display credentials (PhD, LISW, MD, etc) on the bylines for articles published by a person.

Currently they only appear on a person's page and not on any teasers. Screenshots:

Person page (credentials appear):
Screen Shot 2022-04-26 at 1 38 04 PM

Article (credentials do not appear):
Screen Shot 2022-04-26 at 1 52 18 PM

Teaser (credentials do not appear):
Screen Shot 2022-04-26 at 1 37 53 PM

 As I've taken over content editing on a site, I've noticed that my predecessor was getting around this by appending credentials to the person's last names and not utilizing the credentials field:

Screen Shot 2022-04-26 at 1 44 45 PM

# Proposed solution

- Adjust the person teaser to display credentials, or at least add an option to do so
- Possibly adjust the "Credentials" field to include examples and explicitly tell the editor to comma separate multiple credentials. I think some clarity about this field's intended use may be helpful in preventing incorrect use.

Screen Shot 2022-04-26 at 1 48 52 PM

### PR DESCRIPTION
**Pull request by quamsta**
Resolves: https://github.com/uiowa/uiowa/issues/5171

[The trending topics section of the Mental Health at Iowa site](https://mentalhealth.uiowa.edu/trending-topics) has been populated with people that have their credentials stored within their last name. I've been publishing articles for them since April and I've been putting credentials in their last names just to keep them visible in the byline and consistent. It's become apparent that because we're in academia these credentials are important to the authors of posts and I've grown weary of putting them in incorrectly.

This PR adds a person's credentials to the article bylines that appear in article list blocks and on the article itself. 

# How to test

1. Stay on the `main` branch
2. Pull down the Mental Health at Iowa site: `blt ds --site=mentalhealth.uiowa.edu` and login: `drush @mentalhealth.local uli`
3. [Edit Gerard Clancy's person page](https://mentalhealth.uiowa.ddev.site/node/501/edit) to give them proper credentials in the "Credentials" field and remove their improper "last name credentials."
4. Return to the homepage and see Gerard Clancy's credentials are **NOT** showing up in the first post in the "Topics in Mental Health" article list block: Screen Shot 2022-09-29 at 10 56 49 AM
5. Also note that upon clicking the article, their credentials do not show up in the article itself: Screen Shot 2022-09-29 at 10 57 56 AM
6. Check out this branch `fix_article_author_credentials` and clear your cache: `drush @mentalhealth.local cr`
7. Return to the homepage and see Gerard Clancy's proper credentials now appear in the first post in the "Topics in Mental Health" article list block: Screen Shot 2022-09-29 at 10 53 58 AM
8. Also note that their credentials now show up in the article itself: Screen Shot 2022-09-29 at 10 59 36 AM

### CODE REVIEW

File: docroot/modules/custom/sitenow_articles/sitenow_articles.module [hunk 1]
```diff
@@ -98,6 +98,17 @@ function sitenow_articles_preprocess_field(&$variables) {
...
     }
+    // Loop through all author fields in the byline and add credentials
+    // (if available) to their names.
+    foreach ($variables['items'] as $key => $item) {
+      $author_entity = $item['content']['#options']['entity'];
+      if ($author_entity->hasField('field_person_credential') && !$author_entity->get('field_person_credential')
+        ->isEmpty()) {
+        $credentials = $author_entity->field_person_credential->value;
+        $authorTitle = $variables['items'][$key]["content"]["#title"];
+        $variables['items'][$key]["content"]["#title"] = $authorTitle . ", " . $credentials;
```
Comment (pyrello) [inline docroot/modules/custom/sitenow_articles/sitenow_articles.module]: Can we switch these double quotes to single?

Comment (pyrello) [reply docroot/modules/custom/sitenow_articles/sitenow_articles.module]: Also, can we switch `$authorTitle` => `$author_title`?

Comment (pyrello) [reply docroot/modules/custom/sitenow_articles/sitenow_articles.module]: I guess I thought that snake case was the coding standard for Drupal variables, but it appears either is okay: https://www.drupal.org/docs/develop/standards/coding-standards#s-functions-and-variables. In that case, I'm requesting it just to be consistent with how we name variables everywhere else.

Comment (quamsta) [reply docroot/modules/custom/sitenow_articles/sitenow_articles.module]: yeah! no prob on both items. I swear I'd switched camel case -> snake case at some point, but I may have accidentally undid it

e: i did it for author_entity, but not authorTitle. d'oh

Comment (pyrello) [reply docroot/modules/custom/sitenow_articles/sitenow_articles.module]: Also, with PHP8's [nullsafe operator](https://www.php.net/releases/8.0/en.php#nullsafe-operator), I think we might be able to rewrite this as:
```php
$credentials = $author_entity
  ?->field_person_credential
  ?->value;

if (!is_null($credentials)) {
  $variables['items'][$key]['content']['#title'] .= ', ' . $credentials;
}
```

Comment (quamsta) [reply docroot/modules/custom/sitenow_articles/sitenow_articles.module]: Dang, that's some hot-off-the-presses syntax: VS Code hates it, but it does seem to work. Going to do some further testing to ensure it does.

Comment (quamsta) [reply docroot/modules/custom/sitenow_articles/sitenow_articles.module]: @pyrello It does seem to work! Removed the clunky if-statement and replaced with the nullsafe operator. Fun.

**Comment by pyrello** [comment]
I have reviewed this from a code perspective and it looks good after the requested updates! Unfortunately, I can't pull this down and test it because Xcode Developer Tools needs to be updated and I can't get it to complete.

**Comment by quamsta** [comment]
> I have reviewed this from a code perspective and it looks good after the requested updates! Unfortunately, I can't pull this down and test it because Xcode Developer Tools needs to be updated and I can't get it to complete.

@pyrello 
Out of scope for this issue but I had some problems with Xcode recently and had to do it all via commandline:

```
sudo xcode-select --install
sudo xcodebuild -license accept
xcodebuild -runFirstLaunch
```

**Review by pyrello — APPROVED** [review]
Tested and works as advertised. Nice improvement.

**Pull request merged**